common/persistence/visibility/store/elasticsearch/processor.go

```diff
@@ -172,6 +176,15 @@ func (p *processorImpl) hashFn(key interface{}) uint32 {
 
 // Add request to the bulk and return a future object which will receive ack signal when request is processed.
 func (p *processorImpl) Add(request *client.BulkableRequest, visibilityTaskKey string) *future.FutureImpl[bool] {
+	if atomic.LoadInt32(&p.status) == common.DaemonStatusStopped {
+		p.logger.Warn("Skipping ES request for visibility task key because processor has been shut down.", tag.Key(visibilityTaskKey), tag.ESDocID(request.ID), tag.Value(request.Doc))
+		return nil
```

Comment:
Return nil will cause caller to panic I believe. I think we should return a Future, set a processor closed error and return that future.
Skipping request is also inaccurate from my understanding, it's more like the request got rejected.